src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAICrafting.java

```diff
@@ -251,13 +253,13 @@ protected IAIState checkForItems(@NotNull final IRecipeStorage storage)
             {
                 remaining = inputStorage.getAmount();
             }
-            else if(!ItemStackUtils.isEmpty(container) && ItemStackUtils.compareItemStacksIgnoreStackSize(inputStorage.getItemStack(), container , false, true))
+            else if(ItemStackUtils.isEmpty(container) || !ItemStackUtils.compareItemStacksIgnoreStackSize(inputStorage.getItemStack(), container , false, true))
```

Comment:
and again! Here it is even worse, Since here the second part of the OR-Clause has a negation but in the previous cases where you patches this it did not. Why was this changed?

Reply:
reverted